.github/workflows/deploy-docs-and-extensions.yml

```diff
@@ -4,6 +4,11 @@ on:
   push:
     branches:
       - main
+  
+  pull_request:
```

Comment:
i wouldn't recommend deploying the docs workflow on PR requests because GitHub limits write permissions for pull requests coming from forks, you cannot directly push (deploy) in those PR workflows. 
example: had to close this one because deploy docs workflow couldn't run on PR: https://github.com/block/goose/pull/886